app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/relationships/RelationshipPresenterImpl.java

```diff
@@ -134,29 +176,20 @@ public void init(RelationshipContracts.View view) {
                         .subscribeOn(schedulerProvider.io())
                         .observeOn(schedulerProvider.ui())
                         .subscribe(
-                                view.setRelationshipTypes(),
+                                view::setRelationshipTypes,
                                 Timber::e
                         )
         );
     }
 
-    @Override
-    public void goToAddRelationship(String teiTypeToAdd) {
+    public void goToAddRelationship(String teiTypeToAdd) {//need to change to UI
         if (d2.programModule().programs().uid(programUid).blockingGet().access().data().write()) {
             view.analyticsHelper().setEvent(NEW_RELATIONSHIP, CLICK, NEW_RELATIONSHIP);
```

Comment:
analytics helper has a dagger module that can be injected in the presenter constructor. It is better to use that one instead of using the view.